hibernate-core/src/main/java/org/hibernate/internal/util/collections/ConcurrentReferenceHashMap.java

```diff
@@ -451,7 +451,7 @@ final void setValue(V value, ReferenceType valueType, ReferenceQueue<Object> ref
 		}
 
 		@SuppressWarnings("unchecked")
-		static final <K, V> HashEntry<K, V>[] newArray(int i) {
+		static <K, V> HashEntry<K, V>[] newArray(int i) {
```

Comment:
The combination of 'static' and 'final' is really confusing, for the 'final' keyword is only meaningful in non-static context.

Reply:
That is not quite true. A static final is a static that cannot be reassigned.